videostore/utils.go

```diff
@@ -290,3 +295,21 @@ func validateTimeRange(files []string, start, end time.Time) error {
 	}
 	return nil
 }
+
+// getVideoDuration returns the duration of the video file in seconds.
+func getVideoDuration(filePath string) (time.Duration, error) {
+	cFilePath := C.CString(filePath)
+	defer C.free(unsafe.Pointer(cFilePath))
+
+	var duration C.int64_t
+	ret := C.get_video_duration(&duration, cFilePath)
+	switch ret {
+	case C.VIDEO_STORE_DURATION_RESP_OK:
+		// Convert duration from AV_TIME_BASE units to time.Duration
+		return time.Duration(duration) * time.Microsecond, nil
```

Comment:
could duration be zero ever in this case Do we have a test encompassing that case?

Reply:
We should not be getting 0 duration here as I check in the c helper and return an error in that case.

```
  if (*duration == AV_NOPTS_VALUE) {
        avformat_close_input(&fmt_ctx);
        return VIDEO_STORE_DURATION_RESP_ERROR;
    }
```

Will try adding an empty but valid mp4 file in our artifact system to test this out for more confidence.